### PR TITLE
chore: upgrade TypeScript to version 5

### LINKED ISSUE #10156: Update typescript dependency
Update typescript dependency

### PR DESCRIPTION
**Pull request by jonkoops**
Upgrades TypeScript and related dependencies to their respective latest versions.

Closes #10156

### CODE REVIEW

**Comment by nicolethoen** [comment]
I'm game for merging this - but we should definitely test the resulting prerelease in OCP.
They often have issues related to typescript versions when they try to consume PF, since their versions are so much older than PF. 
I'll try to test this locally in OCP before merging, but in case that gets too difficult, don't let my attempts hold this up.

**Review by nicolethoen — APPROVED** [review]

**Comment by jonkoops** [comment]
> but we should definitely test the resulting prerelease in OCP.

Excuse my ignorance, but what is OCP?

**Review by dlabaj — APPROVED** [review]

**Pull request merged**
